src/python/pants/backend/python/rules/pex.py

```diff
@@ -219,7 +219,7 @@ def log(self, *args, **kwargs) -> None:
         self.log_level.log(logger, *args, **kwargs)
 
 
-@rule(name="Create PEX")
+@named_rule(desc="Create PEX")
```

Comment:
The reason the distinction between `@rule` and `@named_rule` makes sense to me is because we need some way of toggling whether a workunit corresponding to a rule gets reported to zipkin/streaming workunits clients. We don't want every single `@rule` to be reported, since many such rules are unimportant. And it's kind of awkward if the distinction between reporting and not reporting a rule workunit is toggled only by whether a description happens to be specified.

Reply:
+1 to having the distinction. I also think it's more discoverable to know that `@named_rule` means a workunit will be created, vs. having to realize that you should pass the `name` parameter to an `@rule` and that that is what distinguishes things.

One idea from this, @gshuflin, is that you can then enforce that every `@named_rule` has a `desc` via kwargs.